pkg/transportrequest/uploadcts.go

```diff
@@ -0,0 +1,227 @@
+
+import (
+	"fmt"
+	"github.com/SAP/jenkins-library/pkg/command"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/piperutils"
+	"strings"
+)
+
+type fileUtils interface {
+	FileExists(string) (bool, error)
+}
+
+var files fileUtils = piperutils.Files{}
+
+// CTSConnection ...
+type CTSConnection struct {
+	Endpoint string
+	Client   string
+	User     string
+	Password string
+}
+
+// CTSApplication ...
+type CTSApplication struct {
+	Name string
+	Pack string
+	Desc string
+}
+
+// CTSNode ...
+type CTSNode struct {
+	DeployDependencies []string
+	InstallOpts        []string
+}
+
+// CTSUploadAction ...
+type CTSUploadAction struct {
+	Connection         CTSConnection
+	Application        CTSApplication
+	Node               CTSNode
+	TransportRequestID string
+	ConfigFile         string
+	DeployUser         string
+}
+
+const (
+	abapUserKey           = "ABAP_USER"
+	abapPasswordKey       = "ABAP_PASSWORD"
+	defaultConfigFileName = "ui5-deploy.yaml"
+)
+
+// WithConnection ...
+func (action *CTSUploadAction) WithConnection(connection CTSConnection) {
+	action.Connection = connection
+}
+
+// WithApplication ...
+func (action *CTSUploadAction) WithApplication(app CTSApplication) {
+	action.Application = app
+}
+
+// WithNodeProperties ...
+func (action *CTSUploadAction) WithNodeProperties(node CTSNode) {
+	action.Node = node
+}
+
+// WithTransportRequestID ...
+func (action *CTSUploadAction) WithTransportRequestID(id string) {
+	action.TransportRequestID = id
+}
+
+// WithConfigFile ...
+func (action *CTSUploadAction) WithConfigFile(configFile string) {
+	action.ConfigFile = configFile
+}
+
+// WithDeployUser ...
+func (action *CTSUploadAction) WithDeployUser(deployUser string) {
+	action.DeployUser = deployUser
+}
+
+// Perform ...
+func (action *CTSUploadAction) Perform(command command.ShellRunner) error {
+
+	command.AppendEnv(
+		[]string{
+			fmt.Sprintf("%s=%s", abapUserKey, action.Connection.User),
+			fmt.Sprintf("%s=%s", abapPasswordKey, action.Connection.Password),
+		})
+
+	cmd := []string{"/bin/bash -e"}
+
+	noInstall := len(action.Node.DeployDependencies) == 0
+	if !noInstall {
+		cmd = append(cmd, "echo \"Current user is '$(whoami)'\"")
+		cmd = append(cmd, getPrepareFioriEnvironmentStatement(action.Node.DeployDependencies, action.Node.InstallOpts))
+		cmd = append(cmd, getSwitchUserStatement(action.DeployUser))
+	} else {
+		log.Entry().Info("No deploy dependencies provided. Skipping npm install call. Assuming current docker image already contains the dependencies for performing the deployment.")
+	}
+
+	deployStatement, err := getFioriDeployStatement(action.TransportRequestID, action.ConfigFile, action.Application, action.Connection)
+	if err != nil {
+		return err
+	}
+
+	cmd = append(cmd, deployStatement)
+
+	return command.RunShell("/bin/bash", strings.Join(cmd, "\n"))
+}
+
+func getPrepareFioriEnvironmentStatement(deps []string, npmInstallOpts []string) string {
+	cmd := []string{
```

Comment:
maybe it would be useful to use the [existing npm package](https://github.com/SAP/jenkins-library/blob/master/pkg/npm/npm.go) for this.